test/DotNetEnv.Tests/EnvTests.cs

```diff
@@ -1,10 +1,51 @@
 ﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
 using Xunit;
 
 namespace DotNetEnv.Tests
 {
-    public class EnvTests
+    public class EnvTests : IDisposable
     {
+        readonly IDictionary<string, string> _oldEnvironment;
+
+        static bool IsWindows => RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
+
+        /// <summary>
+        /// Saves the environment before each test is run.
+        /// </summary>
+
+        public EnvTests() =>
+            _oldEnvironment =
+                Environment.GetEnvironmentVariables()
+                           .Cast<DictionaryEntry>()
+                           .ToDictionary(e => e.Key.ToString(),
+                                         e => e.Value.ToString());
+
+        /// <summary>
+        /// Resets the environment after each test is run.
+        /// </summary>
+
+        public void Dispose()
+        {
+            foreach (var var in _oldEnvironment)
+                Environment.SetEnvironmentVariable(var.Key, var.Value);
+
+            var comparer = IsWindows
+                         ? StringComparer.OrdinalIgnoreCase
+                         : StringComparer.Ordinal;
+
+            var newNames = Environment.GetEnvironmentVariables().Keys
```

Comment:
Why is this needed now?

If it is because of the change from `Fact` to `Theory` then I would prefer just adding a new test case rather than this env var copying

Reply:
It's needed to prove and break the tests in the first place! Without it, the side-effects of one test that may share variables with another will leak.

Whether you use `Fact` or `Theory` is irrelevant. That said, I'm not sure what's there to be gained with duplicating all the code into another test case just to vary the test data (thus the `Theory`).

Reply:
If it avoids the need for this setup and teardown, it gains a lot in my book ;)